app/controllers/portal.js

```diff
@@ -0,0 +1,129 @@
+import Ember from 'ember'
+import generateToken from '../utils/generateToken'
+
+export default Ember.Controller.extend({
+  portalModel: {
+    properties: {
+      username: {
+        type: 'string'
+      }
+    },
+    required: ['username'],
+    type: 'object'
+  },
+  portalView: {
+    containers: [
+      {
+        id: 'main',
+        rows: [
+          [
+            {
+              label: 'GitHub Username',
+              model: 'username'
+            }
+          ]
+        ]
+      }
+    ],
+    rootContainers: [
+      {
+        container: 'main',
+        label: 'Main'
+      }
+    ],
+    type: 'form',
+    version: '1.0'
+  },
+  isFormInvalid: true,
+  actions: {
+    createUser () {
```

Comment:
Also: this function is long. It seems to me that lines 48-55 could be factored into a function. Also, lines 67-75 could be factored into another function. Perhaps those two functions could be factored together?